src/Sql/Auth/dbo/Stored Procedures/AuthRequest_DeleteIfExpired.sql

```diff
@@ -1,6 +1,19 @@
-﻿CREATE PROCEDURE [dbo].[AuthRequest_DeleteIfExpired]
+﻿-- UserExpirationSeconds to 15 minutes (15 * 60)
+-- AdminExpirationSeconds to 7 days (7 * 24 * 60 * 60)
+-- AdminApprovalExpirationSeconds to 12 hour (12 * 60 * 60)
+
+CREATE PROCEDURE [dbo].[AuthRequest_DeleteIfExpired]
+    @UserExpirationSeconds INT = 900,
+    @AdminExpirationSeconds INT = 604800,
+    @AdminApprovalExpirationSeconds INT = 43200
 AS
 BEGIN
     SET NOCOUNT OFF
-    DELETE FROM [dbo].[AuthRequest] WHERE [CreationDate] < DATEADD(minute, -15, GETUTCDATE());
+    DELETE FROM [dbo].[AuthRequest]
+        -- User requests expire after 15 minutes (by default) of their creation
+    WHERE ([Type] != 2 AND DATEADD(second, @UserExpirationSeconds, [CreationDate]) < GETUTCDATE())
+        -- Admin requests expire after 7 days (by default) of their creation if they have not been approved
+        OR ([Type] = 2 AND ([Approved] IS NULL OR [Approved] = 0) AND DATEADD(second, @AdminExpirationSeconds,[CreationDate]) < GETUTCDATE())
+        -- Admin requests expire after 12 hours (by default) of their approval
+        OR ([Type] = 2 AND [Approved] = 1 AND DATEADD(second, @AdminApprovalExpirationSeconds, [ResponseDate]) < GETUTCDATE());
```

Comment:
❓ : when is `ResponseDate` set? 

If it is null in any case, then the DATEADD will error, right? Do we need a null check on `ResponseDate` (e.g., AND [ResponseDate] IS NOT NULL)?

Reply:
`ResponseDate` is set when `Approved` is given a state. https://github.com/bitwarden/server/blob/d020c49c0e3957e4bf08a38be32f306e4225fdda/src/Core/Auth/Services/Implementations/AuthRequestService.cs#L134-L135

So checking that `Approved` is true business logic wise is also me finding out that ResponseDate isn't null. I should add a assertions in #3046 for making sure `ResponseDate` is set on all valid approvals/disapprovals though.

Reply:
Added tests for this in https://github.com/bitwarden/server/pull/3046/commits/95240849fa84b8f18018defe6645f926afda007f.